src/servant/tt_004/sword.c

```diff
@@ -916,9 +913,586 @@ void func_us_801746BC(Entity* self) {
+        D_us_80178B74 = 0;
+        D_us_80170078[0] = self->ext.swordFamiliar.targetX = 0;
+        D_us_8017007C.x = 0;
+        self->ext.swordFamiliar.unk88 = 0x400;
+        g_api.PlaySfx(SFX_ANIME_SWORD_A);
+        self->step++;
+        // fallthrough
+
+    case 4:
+        D_us_80170080.vy += 0x100;
+        if (D_us_80170080.vy >= 0) {
+            D_us_80170080.vy = 0;
+            D_us_801785E4 = 0;
+            g_api.PlaySfx(D_us_80170098[0]);
+            g_PauseAllowed = false;
+            self->step++;
+        }
+        break;
+    case 5:
+        D_us_801785E4++;
+        if (D_us_801785E4 > 0x46) {
+            self->step++;
+        }
+        break;
+
+    case 6:
+        self->ext.swordFamiliar.unk88 += 0x40;
+        self->ext.swordFamiliar.unk88 &= 0xFFF;
+        D_us_8017007C.x = self->ext.swordFamiliar.unk88 - 0x400;
+        D_us_80170080.vz += 0x60;
+
+        if (D_us_80170080.vz >= 0) {
+            D_us_80170080.vz = 0;
+            D_us_801785E8 = 0;
+            self->step++;
+        }
+
+        break;
+
+    case 7:
+        self->ext.swordFamiliar.unk88 += 0x40;
+        self->ext.swordFamiliar.unk88 &= 0xFFF;
+        D_us_8017007C.x = self->ext.swordFamiliar.unk88 - 0x400;
+
+        if (self->ext.swordFamiliar.unk88 == 0xC00) {
+            D_us_801785E4 = 0;
+            self->step++;
+        }
+
+        prim = &g_PrimBuf[self->primIndex];
+        next = prim->next;
+        for (i = 0; i < D_us_801785E8; i++) {
+            next = next->next;
+        }
+
+        last = next->next;
+        *next = *prim;
+        next->next = last;
+        next->clut = 0x142;
+        next->drawMode = DRAW_UNK_100 | DRAW_UNK_40 | DRAW_TPAGE2 | DRAW_TPAGE |
+                         DRAW_COLORS | DRAW_TRANSP;
+        next->r0 = next->r1 = next->r2 = next->r3 = next->g0 = next->g1 =
+            next->g2 = next->g3 = next->b0 = next->b1 = next->b2 = next->b3 =
+                128;
+        D_us_801785E8++;
+        break;
+
+    case 8:
+        D_us_801785E4++;
+        if (D_us_801785E4 > 0xF) {
+            D_us_801785E4 = 0;
+            g_PauseAllowed = true;
+            func_us_801724E8(self, 1, 0U);
+            g_api.PlaySfx(SFX_MAGIC_SWITCH);
+            self->step++;
+        }
+        break;
+
+    case 9:
+        prim = &g_PrimBuf[self->primIndex];
+        next = prim->next;
+        var_s7 = D_us_801785E4 * 32;
+        x0 = prim->x0 - 0x80;
+        x1 = prim->x1 - 0x80;
+        x2 = prim->x2 - 128;
+        x3 = prim->x3 - 128;
+        y0 = prim->y0 - 0x80;
+        y1 = prim->y1 - 0x80;
+        y2 = prim->y2 - 0x80;
+        y3 = prim->y3 - 0x80;
+
+        for (i = 0; i < 64; i++) {
+            s32 temp_s1 = rsin(var_s7);
+            s32 temp_s2 = rcos(var_s7);
+            D_us_801785C4 =
+                ((rcos(var_s7 + 0xC00) * (D_us_801785E4 * 16)) >> 0xC) + 0x80;
+            D_us_801785C8 =
+                ((rsin(var_s7 + 0xC00) * (D_us_801785E4 * 16)) >> 0xC) + 0x80;
```

Comment:
It looks like this might be angle or rotation data, in which case you can use ROT.